pkg/controller/dcl/controller.go

```diff
@@ -205,6 +206,39 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (res
 	if err := r.applyChangesForBackwardsCompatibility(ctx, resource); err != nil {
 		return reconcile.Result{}, fmt.Errorf("error applying changes to resource '%v' for backwards compatibility: %w", k8s.GetNamespacedName(resource), err)
 	}
+
+	cc, ccc, err := resourceactuation.FetchLiveKCCState(ctx, r.mgr.GetClient(), req.NamespacedName)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	am := resourceactuation.DecideActuationMode(cc, ccc)
+	switch am {
+	case v1beta1.Reconciling:
+		r.logger.V(2).Info("Actuating a resource as actuation mode is \"Reconciling\"", "resource", req.NamespacedName)
+	case v1beta1.Paused:
+		jitteredPeriod, err := jitter.GenerateJitteredReenqueuePeriod(r.schemaRef.GVK, nil, r.converter.MetadataLoader, u)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if resource.GetDeletionTimestamp().IsZero() {
+			// add finalizers for deletion defender
```

Comment:
I'm confused as to why we need to touch finalizers here - can you explain?  (Ideally by expanding the comment)

Reply:
I *think* the issue here would be in an uninstallation for KCC case where we don't want the underlying resources on GCP / the cloud provider to go away. So I updated that comment to reflect my understanding.